Cleanup Dockerfile, add support for paho mqtt client_id

WobwobRt opened the pull request:
Hi Antonie,

I've made some changes, primarily to hopefully (it seems, but time will tell) better handle disconnects/reconnects. The entities became unavailable after automated updates of mosquitto. I've only noticed that with mqtt4dsmr. With kamstrup-402-mqtt and zigbee2mqtt this didn't seem to happen, or it got handled better. 

It seems that setting the client_id to a specified value this problem doesn't occur anymore. 

Other  than that, I've further reduced the size of the docker image from 173.1 MB to 81.2 MB. Didn't notice any problems and as far as I can tell (probably/maybe missing something), the image runs fine without cargo, so only the pip install command seems enough. 

Greetings, Arnoud

WobwobRt committed:
Add MQTT_CLIENT_ID handling

WobwobRt committed:
Added MQTT_CLIENT_ID

Hopefully make the connection to mosquitto more robust as this is the only client that doesn't properly reconnect/hanled by mosquitto after an update of mosquitto.

WobwobRt committed:
Dockerfile cleanup

Further reduce docker image size. Workaround doesn't seem necessary anymore. Only copy relevant files instead of full folder.

antonijn commented on Dockerfile:
```diff
@@ -1,20 +1,9 @@
-FROM python:3.13-slim
-WORKDIR /app
-COPY requirements.txt ./
+FROM python:3-alpine
```
python:3-alpine is too unpredictable for my blood. python3.13-alpine is better, I think. Python's version should be fixed as with other dependencies.

antonijn commented on mqtt4dsmr.py:
```diff
@@ -43,7 +43,7 @@ def main():
     cfg = Config()
 
-    client = mqtt.Client(mqtt.CallbackAPIVersion.VERSION2)
+    client = mqtt.Client(mqtt.CallbackAPIVersion.VERSION2,client_id=cfg.MQTT_CLIENT_ID)
```
Interesting, I didn't know about this. I'll hopefully have some time to see how this works later this week/this weekend.

antonijn commented:
Thanks for the PR! Good that you've experimented with the image size. Moving to alpine is probably worthwhile.

WobwobRt committed:
Update config.py MQTT_CLIENT_ID handling

Correction, get_env_opt was missing parameter. Set 'required' to False and default to 'mqtt4dsmr'.

WobwobRt commented:
As the repo has been updated in between, I'll close this request and do and check the changes as well as your recommendations in a new fork. It's been running stable since adding the client_id, but haven't investigated why exactly that makes a difference and whether that's the only and/or correct way.
